src/screen/Home.tsx

```diff
@@ -39,6 +38,15 @@ export default function Home() {
   //   console.log("check stack");
   // };
 
+  const { open } = useWeb3Modal();
+  const navigate = useNavigate();
+
+  const connectWallet = async () => {
+    console.log("wallet connection here");
+    await open({ view: "Networks" });
+    navigate("session");
```

Comment:
navigation bypass wallet connection process, insure to complete wallet connection process first, then navigate to session. In between we have to stack tokes as well.